### ISSUE TITLE
About the usage of Image Picker & problems on SystemChrome.setEnabledSystemUIOverlays([])

### ISSUE THREAD

**Issue by goldenhuang** [flutter/flutter#27890]
I am using Image Picker plugin into my app.
I found that when the following instruction has been issued and returned from the Camera user interface,
```dart
File imagePath = await ImagePicker.pickImage(source: ImageSource.camera);
```
The original setting of 'system UI overlays' has been reset!! 
My app uses `SystemChrome.setEnabledSystemUIOverlays([])` to hide both system status bar and navigation bar but once `ImagePicker.pickImage` is called and it enters Camera UI, sometimes the system status bar and navigation bar cannot be hidden after the return from Camera UI for some reason. 
Is there any way to enforce to recover back the state of system UI overlays after the usage of ImagePicker?

**Comment by zoechi** [comment]
Please add the output of `flutter doctor -v`.

To investigate we would need a minimal runnable reproduction
as a single file so that we can just copy your code into `lib/main.dart` of a new project and run to reproduce.

**Comment by goldenhuang** [comment]
# The following is my output from 'flutter doctor -v' :
[√] Flutter (Channel beta, v1.1.8, on Microsoft Windows [Version 10.0.17763.316], locale zh-TW)
• Flutter version 1.1.8 at C:\Projects\Flutter\flutter
• Framework revision [985ccb6](https://github.com/flutter/flutter/commit/985ccb6d14c6ce5ce74823a4d366df2438eac44f) (6 weeks ago), 2019-01-08 13:45:55 -0800
• Engine revision [7112b72](https://github.com/flutter/flutter/commit/7112b72cc229e05d36716c3d7739885d3ffa72e6)
• Dart version 2.1.1 (build 2.1.1-dev.0.1 ec86471ccc)

[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
• Android SDK at C:\Users\XXXX\AppData\Local\Android\Sdk
• Android NDK location not configured (optional; useful for native profiling support)
• Platform android-28, build-tools 28.0.3
• ANDROID_HOME = C:\Users\XXXX\AppData\Local\Android\Sdk
• Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
• Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
• All Android licenses accepted.

[√] Android Studio (version 3.3)
• Android Studio at C:\Program Files\Android\Android Studio
• Flutter plugin version 32.0.1
• Dart plugin version 182.5124
• Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[√] IntelliJ IDEA Community Edition (version 2018.2)
• IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2018.2.5
• Flutter plugin version 30.0.2
• Dart plugin version 182.4892.25

[√] VS Code
• VS Code at C:\Users\XXXX\AppData\Local\Programs\Microsoft VS Code
• Flutter extension version 2.22.1

[√] VS Code, 64-bit edition (version 1.30.1)
• VS Code at C:\Program Files\Microsoft VS Code
• Flutter extension version 2.22.1

[√] Connected device (1 available)
• LG H830 • LGH830b443707e • android-arm64 • Android 8.0.0 (API 26)

• No issues found!

**Comment by goldenhuang** [comment]
Before applying the instruction "File imagePath = await ImagePicker.pickImage(source: ImageSource.camera);", I have setup my app properly to hide the system status bar and navigation bar by SystemChrome.setEnabledSystemUIOverlays([])... but after calling that instruction and returning from it it seems that somehow my app does not hide those two bars "sometimes"... Sometimes it can hide but sometimes cannot.... (timing issue?)...

**Comment by zoechi** [comment]
To investigate we would need a minimal runnable reproduction
as a single file so that we can just copy your code into `lib/main.dart` of a new project and run to reproduce.

**Comment by goldenhuang** [comment]
The following is my 'build' & 'takePicture' method implementation in the main.dart file. Please refer to it.
It just happened when the user clicked the navigation tab where the code handles different types of tab. 
Just focus on the portion of my takePicture method... 
You can see after calling 'imagePath = await ImagePicker.pickImage(source: ImageSource.camera);' and the camera capture screen comes out and the user can take a photo there... after confirmed and finished, the screen will be back to my main screen page where is supposed to hide the system status bar and the navigation bar at bottom but actually it may or may not hide them with something which depends... 

  @override
  Widget build(BuildContext context) {
    print(MODULE+'=> build');

    // Widget - not _isReady
    //-------------------------------------------------------------------------
    if (!_isReady)
      //  Since _isReady is false, display a circular progress indicator with its valueColor as a fixed color
      return createProgressIndicator();

    // Define the list of widgets resident in BottomNavigationBar
    // Notice that the potential changing named parameter will be passed to the tab so that its didUpdateWidget method will be called once that parameter has been changed
    List<Widget> widgetList = <Widget>[
      MaterialApp(
        home: MapTab(
          category: ScopedModel.of<AppModel>(context, rebuildOnChange: true).mediaCategory,
          markers: ScopedModel.of<AppModel>(context, rebuildOnChange: true).markers,
        ),
        // localizationsDelegates for MaterialApp
        localizationsDelegates: [
          // Put app-specific localization delegate[s] here
          const TranslationsDelegate(),
          GlobalMaterialLocalizations.delegate,
          // can be ignored for non-material app as widget app
          GlobalWidgetsLocalizations.delegate,
        ],
        // supportedLocales for MaterialApp
        supportedLocales: [
          const Locale('en', 'US'),
          // English
          // The following is for Chinese simplified and traditional for different countries
          const Locale.fromSubtags(languageCode: 'zh'),
          const Locale.fromSubtags(languageCode: 'zh', scriptCode: 'Hans'),
          const Locale.fromSubtags(languageCode: 'zh', scriptCode: 'Hant'),
          const Locale.fromSubtags(
              languageCode: 'zh', scriptCode: 'Hans', countryCode: 'CN'),
          const Locale.fromSubtags(
              languageCode: 'zh', scriptCode: 'Hant', countryCode: 'TW'),
          const Locale.fromSubtags(
              languageCode: 'zh', scriptCode: 'Hant', countryCode: 'HK'),
        ],
      ),
      CameraTab(category: ScopedModel.of<AppModel>(context, rebuildOnChange: true).mediaCategory),
      CamcorderTab(category: ScopedModel.of<AppModel>(context, rebuildOnChange: true).mediaCategory),
    ];

    return AnnotatedRegion<SystemUiOverlayStyle>(
      value: SystemUiOverlayStyle.dark,
      child: Scaffold(
        // Display current focused widget in the list
        body: SafeArea(
          top: true,
          bottom: true,
          left: false,
          right: false,
          child: widgetList[_index],
        ),
        // Define the BottomNavigationBar with a Theme object where there is no bottomNavigationBar shown when isOnEULA is on
        bottomNavigationBar: Theme(
          // Define the ThemeData with well-defined canvasColor
          data: ThemeData(canvasColor: Color(0xFF333333)),
          // Define the child of Theme which is the BottomNavigationBar
          child: hideNavigationBar ? Container(height:0, width:0) :BottomNavigationBar(
            // Define current index for the current focused widget
            currentIndex: _index,
            // Define fixedColor for label of the focused widget
            fixedColor: Color(0xFFFFFFFF),
            // Define onTap and set the internal index state here
            onTap: (int index) async {
              // Tap and change navigation bar index (page switching)
              if (this._index == index && index == 0) {
                // Specific handling on MapTab will not be applied across page switching => handle mediaCategory change
                // Advance mediaCategory and update appModel, which will trigger MapTab's didUpdateWidget and thus createAllMarkers will be called there!
                appModel.updateMediaCategory(
                  (appModel.mediaCategory + 1) % MEDIA_CATEGORY_ICONS.length);
              }
              else {
                if (!USE_IMAGE_PICKER) {
                  // For non-specific case just set the state for _index
                  setState(() {
                    this._index = index;
                  });
                  return;
                }
                // The following code is for USE_IMAGE_PICKER only
                if (index == 1) {
                  setState((){
                    hideNavigationBar=true;
                  });
                  await preprocessCameraTab();
                  String filePath = await takePicture();
                  await postprocessCameraTab(filePath);
                  setState((){
                    hideNavigationBar=false;
                  });
                }
                else if (index == 2) {
                  setState((){
                    hideNavigationBar=true;
                  });
                  await preprocessCamcorderTab();
                  String filePath = await recordVideo();
                  await postprocessCamcorderTab(filePath);
                  setState((){
                    hideNavigationBar=false;
                  });
                }
                else;
              }
              // Try to keep the system status bar and navigation bar hidden
              print(MODULE+'=> build: onTap: setEnabledSystemUIOverlays to be []');
              await SystemChrome.setEnabledSystemUIOverlays([]);
            },
            // Define items in the BottomNavigationBar
            items: <BottomNavigationBarItem>[
              // Item - "Map"
              BottomNavigationBarItem(
                icon: genMapIcon(), // generate an ImageIcon widget
                title: genMapTitle(), // generate a Text widget for title
              ),
              // Item - "Camera"
              BottomNavigationBarItem(
                icon: CAMERA_ICON,
                title: Text(Translations.of(context).cameraTab_title, style: TextStyle(color: Color(0xFFFFFFFF))),
              ),
              // Item - "Camcorder"
              BottomNavigationBarItem(
                icon: CAMCORDER_ICON,
                title: Text(Translations.of(context).camcorderTab_title, style: TextStyle(color: Color(0xFFFFFFFF)))
              ),
            ],
          ),
        ),
      )
    );
  }
}

  Future<String> takePicture () async {
    print(MODULE+'=> takePicture: start to take picture');
    // Start to take a picture
    String destPath;
    String filePath;
    File imagePath; // for USE_IMAGE_PICKER only
    try {
      // Prepare destPath in a format from current date time
      var date = DateTime.now();
      filePath = date.year.toString() +
          date.month.toString().padLeft(2, '0') +
          date.day.toString().padLeft(2, '0') +
          '_' +
          date.hour.toString().padLeft(2, '0') +
          date.minute.toString().padLeft(2, '0') +
          date.second.toString().padLeft(2, '0') +
          '.jpg';
      destPath = await MEDIA_PHOTO_PATH + '/IMG_' + filePath;
      // Start to take a photo
      // Retrieve current  file path list in CAPTURED_MEDIA_PATH
      List oldListFiles = Directory(await CAPTURED_MEDIA_PATH).listSync().map((file)=>file.path).toList();
      // Open an ImagePicker with source from the camera, which will return a File object called imagePath
      imagePath = await ImagePicker.pickImage(source: ImageSource.camera);
      if (imagePath == null) {
        // This may happen when either the user cancels the captures (not confirmed OK) or clicks the back tab
        print(MODULE+'=> takePicture: no picture was taken!');
        // For this case we want to delete all of the possible generated temporary files from ImagePicker in the TEMP_MEDIA_PHOTO_PATH folder
        List listFiles = Directory(await TEMP_MEDIA_PHOTO_PATH).listSync();
        for (int i = 0; i < listFiles.length; i++) {
          print(MODULE+'=> takePicture: temp file ${listFiles[i].path} is deleted');
          listFiles[i].deleteSync();
        }
        // Also we need to delete any just captured files (the difference from oldListFiles) in the CAPTURED_MEDIA_PATH folder
        listFiles = Directory(await CAPTURED_MEDIA_PATH).listSync().where((file)=>!oldListFiles.contains(file.path)).toList();
        for (int i = 0; i< listFiles.length; i++) {
          listFiles[i].deleteSync();
          print(MODULE+'=> takePicture: captured file ${listFiles[i].path} is deleted');
        }
        return null;
      }
      print(MODULE + '=> takePicture: rename the captured photo file into the well-defined and preferrable destPath');
      imagePath.renameSync(destPath);
      // Also we need to delete any just captured files (the difference from oldListFiles) in the CAPTURED_MEDIA_PATH folder
      List listFiles = Directory(await CAPTURED_MEDIA_PATH).listSync().where((file)=>!oldListFiles.contains(file.path)).toList();
      for (int i = 0; i< listFiles.length; i++) {
        listFiles[i].deleteSync();
        print(MODULE+'=> takePicture: captured file ${listFiles[i].path} is deleted');
      }
    } catch (e) {
      print(MODULE+'=> takePicture: no picture was taken: '+e.toString());
      return null;
    }
    // Return the file path after taking a picture successfully
    return destPath;
  }

**Comment by zoechi** [comment]
Can you please make it a minimal runnable reproduction
as a single file so that we can just copy your code into `lib/main.dart` of a new project and run to reproduce and ensure it does not contain code that is not required to reproduce the issue.

**Comment by goldenhuang** [comment]
[flutter_app_Feb27_2019.zip](https://github.com/flutter/flutter/files/2873506/flutter_app_Feb27_2019.zip)
Hello,
    Please check the attached zip file which contains the working main.dart and pubspec.yaml files.
    While launching the app, you can see the bottom where there are TWO tabs: one for help and the other one for camera. Initially at the code I have used SystemChrome.setEnabledSystemUIOverlays([]) to enforce the app to hide system status and navigation bar.... But after clicking Camera tab it will enter Camera UI screen, once it is finished and the screen will be back to our main screen; unfortunately it MAY or MAY not keep previous HIDE state even I am trying to add another call to SystemChrome.setEnabledSystemUIOverlays([])  just right after calling ImagePicker... You can try to reproduce it by tapping Camera tab back and forth, the HIDE state is not consistent... Sometimes it can HIDE but sometimes it cannot.

**Comment by cyanglaz** [comment]
@goldenhuang I do want to help but unfortunately, we can't just download the zip and try your code. If you may, could you try to reduce your code **as minimum as** possible to reproduce the issue, and post the main.dart code here?

**Comment by goldenhuang** [comment]
Here is my main.dart file. I have zipped it into main.zip. 
[main.zip](https://github.com/flutter/flutter/files/3124591/main.zip)

**Comment by goldenhuang** [comment]
The following is the post of the main.dart code : 
import 'package:flutter/material.dart';

// => add the code (to use the following plugins)
import 'package:flutter/services.dart';
import 'dart:async';
import 'dart:io';
import 'package:image_picker/image_picker.dart';
// <= end the code

void main() async {
  // =>  add the code (to hide the system status and system navigation bar)
  await SystemChrome.setEnabledSystemUIOverlays([]);
  // <= end the code
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

// class _MyHomePageState extends State<MyHomePage> {
class _MyHomePageState extends State<MyHomePage> with SingleTickerProviderStateMixin {
  int _counter = 0;

  // => add the code
  int _index; // index for page selection
  // Method : initState
  // - Initialize states used in the class
  //---------------------------------------------------------------------------
  @override
  void initState() {
    super.initState();
    _index = 0; // set as page 0
  }
  // <= end the code

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // Define the list of widgets resident in BottomNavigationBar
    List<Widget> widgetList = <Widget>[
      MaterialApp(
        home:  Scaffold(
          appBar: AppBar(
            // Here we take the value from the MyHomePage object that was created by
            // the App.build method, and use it to set our appbar title.
            title: Text(widget.title),
          ),
          body: Center(
            // Center is a layout widget. It takes a single child and positions it
            // in the middle of the parent.
            child: Column(
              // Column is also layout widget. It takes a list of children and
              // arranges them vertically. By default, it sizes itself to fit its
              // children horizontally, and tries to be as tall as its parent.
              //
              // Invoke "debug painting" (press "p" in the console, choose the
              // "Toggle Debug Paint" action from the Flutter Inspector in Android
              // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
              // to see the wireframe for each widget.
              //
              // Column has various properties to control how it sizes itself and
              // how it positions its children. Here we use mainAxisAlignment to
              // center the children vertically; the main axis here is the vertical
              // axis because Columns are vertical (the cross axis would be
              // horizontal).
              mainAxisAlignment: MainAxisAlignment.center,
              children: <Widget>[
                Text(
                  'You have pushed the button this many times:',
                ),
                Text(
                  '$_counter',
                  style: Theme.of(context).textTheme.display1,
                ),
              ],
            ),
          ),
          floatingActionButton: FloatingActionButton(
            onPressed: _incrementCounter,
            tooltip: 'Increment',
            child: Icon(Icons.add),
          ), // This trailing comma makes auto-formatting nicer for build methods.
        ),
      ),
      Container(),
    ];

    return Scaffold(
      body: widgetList[_index],
      bottomNavigationBar: Theme(
        data: ThemeData(canvasColor: Color(0xFF333333)),
        child: BottomNavigationBar(
          currentIndex: _index,
          fixedColor: Color(0xFFFFFFFF),
          onTap: (int index) async {
            if (this._index == index && index == 0) {
              print('Do nothing');
            }
            else {
              if (index == 1) {
                print('Start to enter Camera UI screen');
                // => Need to investigate system status bar and navigation bar from here (HIDE those two bars)
                File imagePath = await ImagePicker.pickImage(source: ImageSource.camera);
                // <= Need to investigate system status bar and navigation bar from here (HIDE or SHOW those two bars for some reasons)
                print('Enable system UI overlays again');
                await SystemChrome.setEnabledSystemUIOverlays([]);
                print('Leave Camera UI screen');
              }
            }
          },
          items: <BottomNavigationBarItem>[
            BottomNavigationBarItem(
              icon: Icon(Icons.help, color: Colors.white),
              title: Text('Help', style:TextStyle(color: Colors.white)),
            ),
            BottomNavigationBarItem(
              icon: Icon(Icons.camera, color: Colors.white),
              title: Text('Camera', style:TextStyle(color: Colors.white)),
            ),
          ],
        ),
      ),
    );
  }
}

**Comment by darshankawar** [comment]
Hi @goldenhuang,
I tried the official `image_picker` plugin example by adding the `SystemChrome.setEnabled..()` and took a picture and selected image from gallery and observed that the setting remains intact, ie, the statusbar and bottom bar remain hidden.

```
void main() {
  WidgetsFlutterBinding.ensureInitialized();
  SystemChrome.setEnabledSystemUIOverlays([]);
  runApp(MyApp());
}
```

[27890.mp4.zip](https://github.com/flutter/flutter/files/5155737/27890.mp4.zip)

Tested on latest stable (1.20.2) and on Android 8.1.0 device.

code sample: https://pub.dev/packages/image_picker/example

Can you confirm at your end if this is still an issue ?
Thanks.

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
